### PR TITLE
Logformatter null check

### PR DESCRIPTION
**Pull request by cijothomas**
`ilogger.Log(LogLevel.Information, default, "Hello World!", null, null);` (note the last param is null, which is the formatter) when combined with `IncludeFormattedMessage=true` will result in an unhandled exception. This PR check if formatter is not null, before invoking it.

Also removed some netcoreapp2.1 checks. (leftover cleanups)

### CODE REVIEW

File: test/OpenTelemetry.Tests/Logs/LogRecordTest.cs [hunk 1]
```diff
@@ -339,6 +322,32 @@ public void IncludeFormattedMessageTest()
...
+        {
+            this.logger.Log(LogLevel.Information, default, "Hello World!", null, null);
+            var logRecord = this.exportedItems[0];
+            Assert.Null(logRecord.FormattedMessage);
+
+            this.options.IncludeFormattedMessage = true;
+            try
+            {
+                // Pass null as formatter function
+                this.logger.Log(LogLevel.Information, default, "Hello World!", null, null);
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Logs/LogRecordTest.cs:336]: @jviau - Please review!

**Review by jviau — APPROVED** [review]
I am not sure this was a bug to begin with. Many other log implementations all explicitly throw on a null formatter.

[ConsoleLogger](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLogger.cs#L42)
[EventLogLogger](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.EventLog/src/EventLogLogger.cs#L77)
[ApplicationInsightsLogger](https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/LOGGING/src/ILogger/ApplicationInsightsLogger.cs#L82)

But with that said, the [aggregate logger](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging/src/Logger.cs) _doesn't_ throw. So I am guessing it is up to each logger to decide what to do.

Either way, this change is fine within context of how otel currently behaves.

File: src/OpenTelemetry/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
 * Removes upper constraint for Microsoft.Extensions.Logging
   dependencies. ([#2179](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2179))
 
+* Fix bug which caused ILogger.Log calls to throw exception, when the
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:8]: Do we consider this as a bug?

File: src/OpenTelemetry/Logs/OpenTelemetryLogger.cs [hunk 3]
```diff
@@ -47,13 +47,22 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
+                    }
+                }
+
                 var record = new LogRecord(
                     options.IncludeScopes ? this.ScopeProvider : null,
                     DateTime.UtcNow,
                     this.categoryName,
                     logLevel,
                     eventId,
-                    options.IncludeFormattedMessage ? formatter(state, exception) : null,
```
Comment (CodeBlanch) [inline src/OpenTelemetry/Logs/OpenTelemetryLogger.cs:56]: Could probably just do:

`options.IncludeFormattedMessage ? formatter?.Invoke(state, exception) : null`

File: src/OpenTelemetry/CHANGELOG.md (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry/CHANGELOG.md:8]: Good qn. After seeing some other implementations, which throw in Log(...formatter=null)., I am thinking we should follow the regular pattern of "not throwing from instrumentation APIs after successful initialization". So Otel logging provider can chose to not throw, end user may still see exception from other providers they have enabled.

And I can reword the changelog to just say:
"OpenTelemetry Logger modified to not throw when formatter is null"

File: src/OpenTelemetry/Logs/OpenTelemetryLogger.cs (hunk 3 above)
Comment (cijothomas) [reply src/OpenTelemetry/Logs/OpenTelemetryLogger.cs:56]: :D yes of course! Will modify.

**Review by reyang — APPROVED** [review]

**Pull request merged**